src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -1001,14 +1004,20 @@ export class EditDataGridPanel extends GridParentComponent {
 
 			applyValue(item, state): void {
 				let activeRow = self.currentCell.row;
-				let currentRow = self.dataSet.dataRows.at(activeRow);
-				let colIndex = self.getColumnIndex(this._args.column.name);
 				let dataLength: number = self.dataSet.dataRows.getLength();
 
 				// If this is not the "new row" at the very bottom
 				if (activeRow !== dataLength) {
-					currentRow[colIndex] = state;
-					this._textEditor.applyValue(item, state);
+					// Prevent applying if the cell contains HTML as it will execute automatically due to slickgrid's design:
+					// the slickgrid text editor's value affects what's in the grid's data, so that means entering HTML will result in execution upon immediate rerender.
+					let containsHTML = /<([A-Za-z][A-Za-z0-9]*)\b[^>]*>(.*?)<\/\1>/.test(state);
```

Comment:
Does this prevent HTML that's already in the database from causing issues?

Reply:
Generally, HTML already in the database will be in the safe cell format that does not execute upon tab switch rerender, so if it is altered in any way by the user, the code will handle it correctly.

Reply:
Generally? When would it not be?

@alanrenmsft @aasimkhan30 You're familiar with this slickgrid stuff, is there any ideas you have here for how to handle this better? Regex to detect HTML and blocking it seems like a not-great user experience.

Reply:
Poor choice of words, it is always in the safe cell format when retrieved directly from the server, it is only immediately after making an edit that the local cell is a not safe format.

Reply:
Gotcha, so as long as the user doesn't actually try to change the value then it's fine? Do you know why the rendering done for that is different than the rendering done when we submit a change?